lib/Doctrine/ORM/Query/Lexer.php

```diff
@@ -164,7 +164,7 @@ protected function getType(&$value)
                 return self::T_STRING;
 
             // Recognize identifiers
-            case (ctype_alpha($value[0]) || $value[0] === '_'):
+            case (ctype_alpha($value[0]) || $value[0] === '_' || $value[0] === '\\'):
```

Comment:
There are no constants that start with `T__` or `T_\\` (the last one is quite un-possible).

I suggest adding more cases for those specifically instead